tests/components/samsungtv/test_media_player.py

```diff
@@ -77,28 +77,30 @@ def tearDown(self):
 
     @MockDependency("samsungctl")
     @MockDependency("wakeonlan")
-    def test_setup(self, samsung_mock, wol_mock):
-        """Testing setup of platform."""
+    async def async_test_setup(self, samsung_mock, wol_mock):
+        """Test setup of platform."""
         with mock.patch("homeassistant.components.samsungtv.media_player.socket"):
             add_entities = mock.Mock()
-            setup_platform(self.hass, WORKING_CONFIG, add_entities)
+            await async_setup_platform(self.hass, WORKING_CONFIG, add_entities)
```

Comment:
Don't set up the platform directly but set up the component that sets up the platform. Use `async_setup_component`.

Let the platform add an entity, then assert the state of the entity by getting the state from the state machine, `hass.states`.

Do this while patching appropriately to avoid I/O.